src/main/java/com/google/sps/data/Riddle.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+// Represents a riddle associated with a destination.
+public class Riddle {
+  private String puzzle;
+  private ArrayList<String> hints;
+
+  public Riddle(String puzzle, ArrayList<String> hints) {
```

Comment:
I think that `List<String>` would be a better parameter type for the hints because it gives the caller more flexibility with what they can pass, e.g. they might have a `LinkedList`.

In general, it's best to specify parameter types that are as general as possible while still enforcing any requirements that you have-- e.g. `List` is good here because it specifies that there's specific sequence of the elements. `Collection` or `Iterable` could kinda-sorta work, but wouldn't be as good of a choice because they don't guarantee that there's a specific ordering. For looking up the specifications of different interfaces and classes, the core library javadocs are the best resource, here's one for `List`: https://docs.oracle.com/javase/8/docs/api/java/util/List.html I usually do a Google search for "list javadoc" or whatever to find these quickly.

Effective Java doesn't really have an item about this, but item 64 is related.